<launch>
	<arg name="drones" default="[1]"/>

    <node pkg="optimal_control_interface" name="solver" type="optimal_control_interface_node" output="screen" ns="uav1">
        <param name="drone_id" value="1"/>
        <param name="solver_rate" value="1"/> <!-- Hz -->
		<rosparam param="drones" subst_value="true">$(arg drones)</rosparam>

      <!-- remapping of topics -->
      <remap from="~odometry_in" to="odometry/odom_main" />
      <remap from="~diagnostics" to="formation_church_planning/diagnostics" />
      <remap from="~desired_trajectory" to="control_manager/trajectory_reference" />
      <remap from="~planned_trajectory" to="formation_church_planning/planned_trajectory" />
      <remap from="~toggle_state" to="formation_church_planning/toggle_state" />
      <remap from="~target_topic" to="/gazebo/dynamic_model/jeff_electrician/odometry" />

    </node>
</launch>

